src/components/Confirm/Confirm.vue

```diff
@@ -1,24 +1,81 @@
 <template>
     <Modal
-        title="Confirm"
-        body="You are about to do something that requires confirmation. Do you wish to continue?"
+        ref="modal"
+        :title="title"
     >
+        <slot>
+            {{ body }}
+        </slot>
+
         <template #footer>
-            <Btn
-                data-bs-dismiss="modal"
-                variant="light"
-            >
-                Cancel
-            </Btn>
-
-            <Btn variant="danger">
-                Confirm
-            </Btn>
+            <slot name="button-cancel">
+                <Button
+                    :outline="cancelOutline"
+                    :variant="cancelVariant"
+                    @click="hide"
+                >
+                    {{ cancelText }}
+                </Button>
+            </slot>
+
+            <slot name="button-ok">
+                <Button
+                    :variant="okVariant"
+                    @click="ok"
+                >
+                    {{ okText }}
+                </Button>
+            </slot>
         </template>
     </Modal>
 </template>
 
-<script lang="ts" setup>
-import Btn from '@/components/Button';
+<script lang="ts">
+import {ComponentPublicInstance, ref} from 'vue';
+import Button from '@/components/Button';
 import Modal from '@/components/Modal';
+import {variantProp} from '@/composables/useVariant';
+</script>
+
+<script lang="ts" setup>
+defineProps({
+    body: {
+        type: String,
+        default: 'Are you sure?',
+    },
+    cancelOutline: {
+        type: Boolean,
+        default: false,
+    },
+    cancelText: {
+        type: String,
+        default: 'No',
+    },
+    cancelVariant: variantProp('light'),
+    okText: {
+        type: String,
+        default: 'Yes',
+    },
+    okVariant: variantProp('primary'),
+    title: {
+        type: String,
+        default: 'Confirmation',
+    },
+});
+
+const emit = defineEmits<{(e: 'confirmed'): void}>();
+
+const ok = () => {
+    emit('confirmed');
+
+    hide();
+};
```

Comment:
Might be useful to have a cancelled emit as well

```suggestion
const emit = defineEmits<{
    (e: 'cancelled'): void,
    (e: 'confirmed'): void,
}>();

const cancel = () => {
    emit('cancelled');

    hide();
};

const ok = () => {
    emit('confirmed');

    hide();
};
```

Reply:
goed idee